apps/web/src/components/AssessmentResultsTable/AssessmentResultsTable.tsx

```diff
@@ -0,0 +1,213 @@
+    const sortedAssessmentSteps = assessmentSteps.sort((a, b) => {
+      if (a.sortOrder && b.sortOrder) {
+        return a.sortOrder - b.sortOrder;
+      }
+
+      if (a.title && b.title) {
+        return Intl.Collator(intl.locale).compare(
+          getLocalizedName(a.title, intl),
+          getLocalizedName(b.title, intl),
+        );
+      }
+
+      return 0;
+    });
+    const columns = sortedAssessmentSteps.reduce(
+      (
+        accumulator: ColumnDef<AssessmentTableRow>[],
+        assessmentStep: AssessmentStep,
+      ) => {
+        const type = assessmentStep.type ?? null;
+        const id =
+          getAssessmentStepType(type ?? "unknownType").id ??
+          uniqueId("results-table-column");
+        const status = columnStatus(assessmentStep, [
+          ...educationResults,
+          ...assessmentResults,
+        ]);
+        const header = columnHeader(
+          intl.formatMessage(getAssessmentStepType(type ?? "unknownType")),
+          status,
+          type,
+          intl,
+        );
+
+        return [
+          ...accumulator,
+          buildColumn({
+            id,
+            header,
+            poolCandidate,
+            assessmentStep,
+            intl,
+          }),
+        ];
+      },
+      [],
+    );
+
+    return [
+      // Insert skills header at top of columns order
+      columnHelper.accessor(
+        ({ poolSkill }) => getLocalizedName(poolSkill?.skill?.name, intl),
+        {
+          id: "requirement",
+          header: intl.formatMessage({
+            defaultMessage: "Requirement",
+            id: "jAWP0X",
+            description:
+              "Header for requirement section of assessment results table",
+          }),
+          cell: ({ row: { original } }) =>
+            cells.jsx(
+              <span>
+                {original.poolSkill ? (
+                  <>
+                    <span data-h2-font-weight="base(bold)">
+                      {getLocalizedName(original.poolSkill?.skill?.name, intl)}{" "}
+                    </span>
+                    <span>
+                      (
+                      {intl.formatMessage(
+                        original.poolSkill?.type
+                          ? getPoolSkillType(original.poolSkill.type)
+                          : commonMessages.notFound,
+                      )}
+                      )
+                    </span>
+                  </>
+                ) : (
+                  <span data-h2-font-weight="base(bold)">
+                    {intl.formatMessage({
+                      defaultMessage: "Education requirement",
+                      id: "4xXPIe",
+                      description: "Education requirement row header.",
+                    })}
+                  </span>
+                )}
+              </span>,
+            ),
+        },
+      ) as ColumnDef<AssessmentTableRow>,
```

Comment:
Why is the cast necessary here?

Reply:
That's a good question 😆. It has to with property `accessorFn` missing otherwise. I'm looking at the docs and it looks like it should be able to [accept undefined](https://tanstack.com/table/v8/docs/api/core/column-def#accessorfn). We do this on all our tables. I think it was intentionally built this way, maybe Eric might know more since he built it.

Reply:
heh, I honestly cannot remember 😆 I just removed some of them and everything seems to be fine 🤔 

this may have been a case where I was seeing errors and the cast was a temporary measure put in place to get them to go away and I just forgot to remove it and it became a pattern we all followed? 😬

Reply:
Okay, if it works without it, lets remove it. Casting should only be a last resort, since it can hide real type errors!

Reply:
I'll make a new issue for this since it spans across multiple tables!